neutrino.go

```diff
@@ -1078,6 +1078,33 @@ func (s *ChainService) BanPeer(addr string, reason banman.Reason) error {
 	return s.banStore.BanIPNet(ipNet, reason, BanDuration)
 }
 
+// UnBanPeer connects and unban's a previously banned peer.
+func (s *ChainService) UnBanPeer(addr string) error {
+	log.Warnf("UnBanning peer %v", addr)
+
+	defer func() {
+
+		if sp := s.PeerByAddr(addr); sp != nil {
```

Comment:
I don't think it's necessary to also trigger a connection request in the background. Instead it should be enough to simply unban the peer. 

This also has a race condition: if this goroutine execute before the logic below, then we'll connect out to the peer but then tear down the connection as it's from a banned peer.

Reply:
Thanks  a lot for the review but I would like to please clarify what goroutine that you are referring to. As I did not include a goroutine in the function. Are you saying that I should remove from line 1085-1098?

Reply:
Ah, I scanned over it quickly and didn't realize this was actually just a `defer`. Since the unban call is sync, this should be ok.